python-package/lightgbm/dask.py

```diff
@@ -505,7 +501,7 @@ def _predict(
             pred_leaf=pred_leaf,
             pred_contrib=pred_contrib,
             dtype=dtype,
-            **kwargs
+            drop_axis=1
```

Comment:
Hi James. This was necessary for `predict_proba` with `raw_score=True` for binary classification, because the current handling of this is:
```python
if pred_proba:
    kwargs['chunks'] = (data.chunks[0], (model.n_classes_,))
else:
    kwargs['drop_axis'] = 1
```
however, for binary classification with raw_score the result doesn't have `model.n_classes_` (2) columns but just 1. So I had to include more conditions to check if this was the raw_score of binary classification. I messed up the conditions once and saw that it went to the else and it worked. Same for the dataframe case where I changed it to just check the shape of the `result` array.

Reply:
ah now I understand better what you meant in the PR description. Sounds good, thanks very much for the fix!